sktime/transformations/panel/dictionary_based/_sfa.py

```diff
@@ -163,7 +159,9 @@ def __init__(
         self.skip_grams = skip_grams
 
         self.return_pandas_data_series = return_pandas_data_series
-        self.fourier_transform = fourier_transform
+        self.fourier_transform = (
+            fourier_transform if word_length < window_size - offset else "dft"
```

Comment:
I really like this change, thank you :)

Just one minor thing: currently `self.fourier_transform` is a string with a check: `self.fourier_transform == "dft"`

But as this is always a boolean condition with True, False, it would be better to use it as a boolean? 

Maybe something like:
`self.use_fallback_dft=True | False`

Reply:
From the results I have run the numba DFT seems faster than the numpy FFT anyway, but I expected some pushback with fully replacing it. Just keeping one would solve some of the issues.

If not that's a more sensible way of handling it.

Reply:
Just renamed and refactored to use booleans for now.

Reply:
> From the results I have run the numba DFT seems faster than the numpy FFT anyway, but I expected some pushback with fully replacing it. Just keeping one would solve some of the issues.
> 
> If not that's a more sensible way of handling it.

I does makes sense, when using the first few coefficients of the DFT, as only `output_length` iterations have to be performed, and the complexity is O(output_length * n).

It might be different, when output_length becomes large, as for example with the supervised transform in WEASEL, as in that case `output_length` might become larger than log n, and thus the numpy fft is faster with O( n log n)

Reply:
So, as a future extension, one might use the numba DFT when `anova=False` and the numpy DFT when `anova=True`